golem-worker-executor/src/worker/mod.rs

```diff
@@ -1572,15 +1572,47 @@ impl RunningWorker {
                     },
                 );
 
-            parent
+            match parent
                 .component_service()
                 .get(
                     &parent.engine(),
                     &account_id,
                     &component_id,
                     component_version,
                 )
-                .await?
+                .await
+            {
+                Ok((component, component_metadata)) => Ok((component, component_metadata)),
+                Err(error) => {
+                    if component_version != worker_metadata.last_known_status.component_version {
```

Comment:
Does it make sense to inspect the error here and look for an explicit not_found?

Imo for infrastructure errors it would be better to not fail the update and retry it whenever the component service comes back up. Plus the following get (for the current version) will fail in the case of infrastructure errors anyway, and we need to handle it.

Reply:
I wanted that first but then found that out here isn't a specific error case for this and didn't want to change GolemError as I know you were also looking into that.